zerver/lib/thumbnail.py

```diff
@@ -437,18 +442,18 @@ def rewrite_thumbnailed_images(
 
         image_data = images.get(path_id)
         if image_data is None:
-            # Has not been thumbnailed yet; leave it as a spinner.
-            # This happens routinely when a message contained multiple
-            # unthumbnailed images, and only one of those images just
-            # completed thumbnailing.
+            # The message has multiple images, and we're updating just
+            # one image, and it's not this one.  Leave this one as-is.
+            remaining_thumbnails.add(path_id)
+        elif image_data.url is None:
+            # We're re-rendering the whole message, so fetched all of
+            # the image metadata rows; this is one of the images we
+            # about, but is not thumbnailed yet.
             remaining_thumbnails.add(path_id)
         else:
             changed = True
             del image_tag["class"]
             image_tag["src"] = image_data.url
-            image_tag["data-original-dimensions"] = (
-                f"{image_data.original_width_px}x{image_data.original_height_px}"
-            )
```

Comment:
Hmm, I guess if we had any pre-patch outstanding loading spinners at the time that this PR is deployed on a given server, we'd need this line in order to make sure the dimensions get added?

Reply:
Mmm, true.  There's little cost to adding it back.